wxgui-src/wxmain.cc

```diff
@@ -103,6 +103,9 @@ struct GameApp : public wxApp
     /* The "shadow" game state for displaying to the user. */
     std::vector<SystemWidget *> stars;
 
+    bool autosave = false;
+    char filename[1024];
```

Comment:
Please resubmit this patch with `filename` being a `std::string`. I don't see any reason to limit the user's filename to 1024 characters, nor any reason to use 1024 bytes to store the filename if it happens to be, say, `"foo.txt"`.